resource-managers/kubernetes/docker/src/main/dockerfiles/spark/decom.sh

```diff
@@ -18,17 +18,24 @@
 #
 
 
-set -ex
+set +e
+set -x
 echo "Asked to decommission"
 # Find the pid to signal
 date | tee -a ${LOG}
-WORKER_PID=$(ps -o pid -C java | tail -n 1| awk '{ sub(/^[ \t]+/, ""); print }')
+WORKER_PID=$(ps x -o pid,cmd -C java |grep Executor \
```

Comment:
This did not return the correct process to kill for me

Reply:
Intresting, can you send me the ps aux for the container?

Reply:
We could also do a `killall java` instead but I'm hesistant incase someones runing another java process inside the executor container.

Reply:
Here it is:

<img width="1626" alt="image" src="https://user-images.githubusercontent.com/30076090/176118840-7681517d-b53b-4c50-a1b9-7ca119f2e05d.png">

Tailing the last line seems to grab the process running `grep executor` rather than the actual executor process

Reply:
interesting, the `-C java` should filter it down to just java processes.

Reply:
Ah I figured it out, the `ps x` is overriding the `-C' for you. I'll fix that.

Reply:
Great, that gets the correct process now